src/transport/HighPressureGasTransport.cpp

```diff
@@ -327,167 +323,579 @@ double HighPressureGasTransport::viscosity()
+    vector<double> sigma_i(nsp), epsilon_over_k_i(nsp), acentric_factor_i(nsp), MW_i(nsp), kappa_i(nsp);
+    for (size_t i = 0; i < m_nsp; i++) {
+        // From equation 9-5.32.
+        double m3_per_kmol_to_cm3_per_mol = 1e3; // Convert from m^3/kmol to cm^3/mol
+        double Vc = Vcrit_i(i) * m3_per_kmol_to_cm3_per_mol;
+        sigma_i[i] = 0.809*pow(Vc, 1.0/3.0);
+
+        // From equation 9-5.34.
+        epsilon_over_k_i[i] = Tcrit_i(i)/1.2593;
+
+        // NOTE: The association parameter is assumed to be zero for all species, but
+        // is left here for completeness or future revision.
+        kappa_i[i] = 0.0;
+
+        // These values are available from the base class
+        acentric_factor_i[i] = m_w_ac[i];
+        MW_i[i] = m_mw[i];
+    }
+
+    // Here we use the combining rules defined on page 9.25.
+    // We have ASSUMED that the binary interaction parameters are unity for all species
+    // as was done in the Chung method.
+    //
+    // The sigma & kappa relations can be fully computed in the loop. The other ones require a final
+    // division by the final mixture values, and so the quantities in the loop are only the
+    // numerators of the equations that are referenced in the comments. After the loop, the
+    // final mixture values are computed and stored.
+    for (size_t i = 0; i < m_nsp; i++) {
+        for (size_t j = 0; j <m_nsp; j++){
+            double sigma_ij = sqrt(sigma_i[i]*sigma_i[j]); // Equation 9-5.33
+            params.sigma_mix += molefracs[i]*molefracs[j]*pow(sigma_ij,3.0); // Equation 9-5.25
+
+            double epsilon_over_k_ij = sqrt(epsilon_over_k_i[i]*epsilon_over_k_i[j]); // Equation 9-5.35
+            params.epsilon_over_k_mix += molefracs[i]*molefracs[j]*epsilon_over_k_ij*pow(sigma_ij,3.0); // Equation 9-5.27 (numerator only)
+
+            // This equation is raised to the power 2, so what we do here is first store only the
+            // double summation into this variable, and then later square the result.
+            double MW_ij =  (2*MW_i[i]*MW_i[j]) / (MW_i[i] + MW_i[j]); // Equation 9-5.40
+            params.MW_mix += molefracs[i]*molefracs[j]*epsilon_over_k_ij*pow(sigma_ij,2.0)*sqrt(MW_ij); // Equation 9-5.28 (double summation only)
+
+            double acentric_factor_ij = 0.5*(acentric_factor_i[i] + acentric_factor_i[j]); // Equation 9-5.36
+            params.acentric_factor_mix += molefracs[i]*molefracs[j]*acentric_factor_ij*pow(sigma_ij,3.0); // Equation 9-5.29
+
+            // The base class' dipole moment values are in the SI units, so we need to convert to
+            // Debye units.
+            double SI_to_Debye = 1.0 / 3.335e-30; // Conversion factor from C*m to Debye
+            double dipole_ii = m_dipole(i,i)*SI_to_Debye;
+            double dipole_jj = m_dipole(j,j)*SI_to_Debye;
+            params.mu_mix += molefracs[i]*molefracs[j]*pow(dipole_ii,2.0)*pow(dipole_jj,2.0)/pow(sigma_ij,3.0); // Equation 9-5.30
```

Comment:
Equivalently, with one fewer call to `pow`:
```suggestion
            params.mu_mix += molefracs[i]*molefracs[j]*pow(dipole_ii*dipole_jj,2.0)/pow(sigma_ij,3.0); // Equation 9-5.30
```